q2cli/core/usage.py

```diff
@@ -74,14 +75,18 @@ def to_interface_name(self):
 
         return cli_name
 
-    def assert_has_line_matching(self, path, expression):
+    def assert_has_line_matching(self, path, expression, key=None):
         if not self.use.enable_assertions:
             return
 
         INDENT = self.use.INDENT
         input_path = self.to_interface_name()
         expr = shlex.quote(expression)
 
+        if key:
+            input_path = \
+                "%s%s%s" % (input_path, key, self.execute()[key].extension)
+
```

Comment:
@lizgehret I realized something after you left. 

1. I lied and all elements in the collection might not be the same type. It will be for an output, but not necessarily for an input. 

2. We have the dict. We can get the item we want from the dict. We should probably just get that item and read its extension and not do the goofy thing we were doing before.